### PR TITLE
Create a simple e2e test

### PR DESCRIPTION
**Pull request by schjan**
Also remove custom Registry Client as it is buggy when it comes to authentication.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -36,3 +36,7 @@
 [[constraint]]
   branch = "master"
   name = "github.com/giantswarm/backoff"
+
+[[constraint]]
+  branch = "master"
+  name = "github.com/nokia/docker-registry-client"
```
Comment (xh3b4sd) [inline Gopkg.toml]: Why don't we use `github.com/docker/docker`?

Comment (schjan) [reply Gopkg.toml]: docker/docker is a client for your local docker instance.
Sadly there is no offical API client for Docker Registries.

**Comment by schjan** [comment]
What do you think shoud I create a aliyun e2e test in this PR, too?

File: Gopkg.toml (hunk 1 above)
Comment (xh3b4sd) [reply Gopkg.toml]: What is with https://godoc.org/github.com/docker/distribution/registry/client?

Comment (schjan) [reply Gopkg.toml]: I don't see how this implements our needed functionality.

**Pull request merged**